pkg/controller/proxyconfig/proxyconfig_controller.go

```diff
@@ -0,0 +1,141 @@
+	reconcilePeriod = 30 * time.Second
+	reconcileResult = reconcile.Result{RequeueAfter: reconcilePeriod}
+)
+
+// Add creates a new ClusterLogging Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	if err := configv1.Install(mgr.GetScheme()); err != nil {
+		return &ReconcileProxyConfig{}
+	}
+
+	return &ReconcileProxyConfig{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("proxyconfig-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to the additional trust bundle configmap.
+	if err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return err
+	}
+
+	// Watch for changes to the proxy resource.
+	if err = c.Watch(&source.Kind{Type: &configv1.Proxy{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileProxyConfig{}
+
+// ReconcileProxyConfig reconciles a ClusterLogging object
+type ReconcileProxyConfig struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a cluster-scoped named "cluster" as well as
+// trusted CA bundle configmap objects for the collector and the visualization resources.
+// When the user configured and/or system certs are updated, the change is propagated to the
+// configmap objects and this reconciler triggers to restart those pods.
+func (r *ReconcileProxyConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	loggingNamespacedName := types.NamespacedName{Name: constants.SingletonName, Namespace: constants.OpenshiftNS}
+	proxyNamespacedName := types.NamespacedName{Name: constants.ProxyName}
+	var proxyConfig *configv1.Proxy = nil
+	var trustBundle *corev1.ConfigMap = nil
+	if request.NamespacedName == proxyNamespacedName {
+		proxyConfig = &configv1.Proxy{}
+		if err := r.client.Get(context.TODO(), request.NamespacedName, proxyConfig); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Request object not found, could have been deleted after reconcile request.
+				// Return and don't requeue
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - just return without requeuing.
+			return reconcile.Result{}, err
+		}
+	} else if request.Name == constants.SingletonName {
```

Comment:
it looks like this is trying to determine if the reconcile request was triggered by a a change to a configmap that logging cares about, but i'm not sure how this works?

Reply:
The above code is "Outdated".
Now we are checking the request.Name is in ReconcileForGlobalProxyList, which is {"fluentd-trusted-ca-bundle", "kibana-trusted-ca-bundle"}
`} else if utils.ContainsString(constants.ReconcileForGlobalProxyList, request.Name) {`
https://github.com/openshift/cluster-logging-operator/pull/255/files#diff-993a3a5d79e9b1a8103e2bde12087a84R108